site/_posts/2020-07-23-securing-react-native-with-oauth.md

```diff
@@ -0,0 +1,480 @@
+
+{% include _image.liquid src="/assets/img/blogs/react-native-oauth/fusionauth-dashboard.png" alt="Configuring the FusionAuth application." class="img-fluid" figure=false %}
+
+Make sure to register your user to the new application. If you want, you can add more users; register them as well.
+
+{% include _image.liquid src="/assets/img/blogs/react-native-oauth/fusionauth-dashboard-register-user.png" alt="Registering your user to the React Native FusionAuth application." class="img-fluid" figure=false %}
+
+Now, we move on to the React Native project.
+
+## Setting up the React Native development environment
+
+Since we are going to use the React Native command line interface tool (CLI) for development, we must have the React Native development environment installed. For installation instructions, please follow [the official documentation](https://reactnative.dev/docs/environment-setup).
+
+We also need to install development environments for iOS, Android, or both. We are going to use [brew](https://brew.sh/) to install additional packages. Make sure that brew is already installed, or install the packages in a different way. 
+
+### iOS environment
+
+Install needed a needed iOS dependency using this command:
+
+```shell
+brew install watchman
+```
+
+We need to install the Xcode CLI tools, which are not normally present. To do so, open Xcode and navigate to "Preferences" and then "Locations". Then pick the Xcode version for command-line tools as shown in the screenshot below:
+
+{% include _image.liquid src="/assets/img/blogs/react-native-oauth/activate-xcode.png" alt="Activating Xcode." class="img-fluid" figure=false %}
+
+The setup for iOS is now complete. 
+
+### Android environment
+
+For Android, JDK 8 is required. Using other versions may result in errors. We can download this JDK from the Oracle website or using brew as shown in the snippet below:
+
+```shell
+brew cask install adoptopenjdk/openjdk/adoptopenjdk8
+```
+
+Next, we need to download and install the [Android studio](https://developer.android.com/studio/install).
+
+Then, we need to configure the `ANDROID_HOME` environment variable in our system path.  We can add the following lines to our `$HOME/.bash_profile` or `$HOME/.bashrc`. If you are using zsh then the files are `~/.zprofile` or `~/.zshrc`.
+
+```bash
+export ANDROID_HOME=$HOME/Library/Android/sdk
+export PATH=$PATH:$ANDROID_HOME/emulator
+export PATH=$PATH:$ANDROID_HOME/tools
+export PATH=$PATH:$ANDROID_HOME/tools/bin
+export PATH=$PATH:$ANDROID_HOME/platform-tools
+```
+
+Now our setup for the Android platform is complete.
+
+## React native project setup
+
+We are now going to create a new React Native project. First, create a new project in our desired directory by running the following command in the terminal:
+
+```shell
+react-native init RNfusionauth
+```
+
+Open the project folder in your text editor, as we'll be making additional changes to these files.
+
+### Installing react native app auth
```

Comment:
"Installing react native app auth for communicating with OAuth 2.0 and OpenId Connect"